actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -453,8 +453,10 @@ def handle_options(options)
 
           options[:path]      ||= "/"
 
-          cookies_same_site_protection = request.cookies_same_site_protection
-          options[:same_site] ||= cookies_same_site_protection.call(request)
+          unless options[:same_site] == false
+            cookies_same_site_protection = request.cookies_same_site_protection
+            options[:same_site] ||= cookies_same_site_protection.call(request)
+          end
```

Comment:
Would it make more sense to just not overwrite `:same_site` if it exists?

```suggestion
          unless options.key?(:same_site)
            options[:same_site] = request.cookies_same_site_protection.call(request)
          end
```

That would permit specifying `nil`.